### PR TITLE
Enhancement/cleanup

### PR DESCRIPTION
**Pull request by swashcap**
* **Problem:** steelpenny development practices made the project confusing to reason about and contribute to
* **Solution:** Address linting problems, directory structure, old tooling, etc.
* **Testing:** Examine commit-by-commit. Ensure things look :+1:. Run linting and testing.

### CODE REVIEW

File: packages/steelpenny/.nycrc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (swashcap) [inline packages/steelpenny/.nycrc]: We were using an _old_ version of istanbul. nyc’s the _hotness_, and it’s much easier to configure in the dotfile compared to the npm script inside _package.json_.

File: packages/steelpenny/package.json [hunk 2]
```diff
@@ -71,23 +68,23 @@
...
     "replace": "0.3.0",
     "should": "7.1.0",
     "sinon": "1.17.3",
     "webpack": "1.12.14"
   },
   "scripts": {
     "build": "node scripts/build.js",
-    "test": "npm run build && istanbul cover -x '**/dist/**' ${PWD}/node_modules/.bin/_mocha -- --timeout=10000 -R spec 'src/test/integration/*.js' 'src/test/unit/*.js'",
-    "postest": "istanbul check --statements 90 --functions 90",
+    "test": "npm run build && nyc mocha --timeout=10000 -R spec 'test/{integration,unit}/*.js'",
```
Comment (swashcap) [inline packages/steelpenny/package.json]: I imagine nyc addressed a bug that required us to use the confusing path junk with istanbul.

File: packages/steelpenny/package.json [hunk 3]
```diff
@@ -101,8 +98,8 @@
     "docs-client": "mkdir -p ../halfpenny/docs && jsdoc -c .jsdoc.json -t ./node_modules/minami -d ../halfpenny/docs -r src/client/ -r -R ../halfpenny/README.md dist/codegen-client-compiled/src",
     "deploy-docs": "gh-pages -d docs && rm -rf docs",
     "postpublish": "npm run docs && npm run deploy-docs",
-    "lint": "eslint src/ test/",
-    "lintfix": "eslint --fix src/ test/"
+    "lint": "eslint *.js scripts src test",
```
Comment (swashcap) [inline packages/steelpenny/package.json]: Gotta lint those scripts and webpack file.

File: packages/steelpenny/package.json [hunk 4]
```diff
@@ -116,6 +113,7 @@
   "homepage": "https://github.com/MRN-Code/quarterback",
   "pre-commit": [
     "validate",
+    "lint",
```
Comment (swashcap) [inline packages/steelpenny/package.json]: 🎉🙌🍾

File: packages/steelpenny/scripts/generate-client.js [hunk 5]
```diff
@@ -31,37 +31,39 @@ const codegenTemplatePath = path.join(
...
+  `-i ${swaggerSpecPath}`,
+  '--lang javascript',
+  `-t ${codegenTemplatePath}`,
+  `--output ${clientDestPath}`,
 ].join(' ');
 
 const swaggerSpecUrl = '/swagger/swagger.json';
 
+/* eslint-disable no-console */
 const logInfo = (msg) => console.log(chalk.white(msg));
```
Comment (swashcap) [inline packages/steelpenny/scripts/generate-client.js]: There’s a lot of interesting `console.log` calls that we should examine and potentially move to a winston-wrapped logger. Another PR, perhaps.

File: packages/steelpenny/src/client/authClient.js [hunk 6]
```diff
@@ -13,16 +13,16 @@ const authClient = {
...
+   * @param  {object} authStorage the localstorage provider to use for creds
+   * @returns {object}             the authClient
+   */
   init: (apiClient, config) => {
-    var authStorage = config.store;
-    var agent = config.agent;
-    apiClient.auth = authClient;
+    const authStorage = config.store;
+    const agent = config.agent;
+    apiClient.auth = authClient; // eslint-disable-line no-param-reassign
```
Comment (swashcap) [inline packages/steelpenny/src/client/authClient.js]: This whole module is pretty confusing due to strange mutation/reassignment. Needs some examination.

File: packages/steelpenny/src/lib/app-routes/coins/scan-details.js [hunk 7]
```diff
@@ -1,56 +0,0 @@
-'use strict';
```
Comment (swashcap) [inline packages/steelpenny/src/lib/app-routes/coins/scan-details.js]: All these these route files were moved from _app-routes_ to _routes_. GitHub isn't picking up on this?

File: packages/steelpenny/src/lib/controllers/reset.js [hunk 8]
```diff
@@ -73,16 +82,10 @@ internals.sendMail = (Mail, resetData) => {
     subject: 'COINS Password Reset',
     templateLocals: {
       html: {
-        messageBody: ['Your COINS account password has been requested ',
-                `to be reset. Use this link to reset it: ${link}`,
-                '<br>If this message is in error, you can safely ignore it.']
-                .join('')
+        messageBody: messageBody.replace(/\n\n/g, '<br>'),
```
Comment (swashcap) [inline packages/steelpenny/src/lib/controllers/reset.js]: A little workaround to make things a bit prettier. Replace those double line breaks with `<br>`s for the HTML email.

File: packages/steelpenny/src/lib/models/CasRole.js [hunk 9]
```diff
@@ -1,11 +1,10 @@
...
-module.exports = function (bookshelf) {
-  return bookshelf.extend({
-    tableName: 'cas_roles',
-    idAttribute: 'role_id',
-    user: function () {
-      return this.belongsTo('UserStudyRole', 'role_id');
-    }
-  });
-};
+module.exports = bookshelf => bookshelf.extend({
```
Comment (swashcap) [inline packages/steelpenny/src/lib/models/CasRole.js]: Easier to write these factories as arrow functions.

File: packages/steelpenny/src/lib/utils/authentication.js [hunk 10]
```diff
@@ -14,189 +14,180 @@ const moment = require('moment');
...
-          callback(null, credentials);
-        }
-    });
-  };
+  /**
+   * helper function to revoke all roles from Relations
+   * @param {string} username - the username to retrieve roles
+   * @return {object} roles - the current roles
+   */
+  function revokeRoles(username) {
```
Comment (swashcap) [inline packages/steelpenny/src/lib/utils/authentication.js]: A little reordering so that the functions are defined before called.

Comment (swashcap) [inline packages/steelpenny/src/lib/utils/authentication.js]: This is a linter rule, btw.

File: packages/steelpenny/test/integration/response-formatter.js [hunk 11]
```diff
@@ -117,27 +117,27 @@ describe('responseFormatter', () => {
...
-      method: 'GET'
+      method: 'GET',
     };
 
-    return server.injectThen(request). then((response) => {
+    return server.injectThen(request).then((response) => {
       verifyResponseFormat(response.result);
       response.statusCode.should.equal(200);
-      expect(response.result.error).to.be.null; // jshint ignore:line
+      expect(response.result.error).to.be.null; // eslint-disable-line no-unused-expressions
```
Comment (swashcap) [inline packages/steelpenny/test/integration/response-formatter.js]: Is there a BDD flag for ESLint?

**Comment by swashcap** [comment]
Looks like I need to fix some merge conflicts…😱

**Comment by swashcap** [comment]
Also, I targeted _dcrerwrite_ as the base branch since it's essentially the active development branch. Should I change the base to _master_ and rebase?

File: packages/steelpenny/scripts/generate-client.js (hunk 5 above)
Comment (rssk) [inline packages/steelpenny/scripts/generate-client.js]: Double toats

File: packages/steelpenny/package.json [hunk 12]
```diff
@@ -101,8 +101,8 @@
     "docs-client": "mkdir -p ../halfpenny/docs && jsdoc -c .jsdoc.json -t ./node_modules/minami -d ../halfpenny/docs -r src/client/ -r -R ../halfpenny/README.md dist/codegen-client-compiled/src",
     "deploy-docs": "gh-pages -d docs && rm -rf docs",
     "postpublish": "npm run docs && npm run deploy-docs",
-    "lint": "eslint src/ test/",
-    "lintfix": "eslint --fix src/ test/"
+    "lint": "eslint *.js scripts src",
```
Comment (rssk) [inline packages/steelpenny/package.json]: Why not test?

File: packages/steelpenny/package.json (hunk 4 above)
Comment (rssk) [inline packages/steelpenny/package.json]: 👍

File: packages/steelpenny/package.json [hunk 13]
```diff
@@ -101,8 +101,8 @@
     "docs-client": "mkdir -p ../halfpenny/docs && jsdoc -c .jsdoc.json -t ./node_modules/minami -d ../halfpenny/docs -r src/client/ -r -R ../halfpenny/README.md dist/codegen-client-compiled/src",
     "deploy-docs": "gh-pages -d docs && rm -rf docs",
     "postpublish": "npm run docs && npm run deploy-docs",
-    "lint": "eslint *.js scripts src",
-    "lintfix": "eslint --fix *.js scripts src"
+    "lint": "eslint *.js scripts src test",
```
Comment (rssk) [inline packages/steelpenny/package.json]: ah, jk

File: packages/steelpenny/package.json [hunk 14]
```diff
@@ -43,16 +43,14 @@
     "moment": "2.10.6",
     "newrelic": "1.26.1",
     "nomnom": "1.8.1",
-    "pg": "4.4.2",
+    "pg": "6.0.0",
```
Comment (rssk) [inline packages/steelpenny/package.json]: holy version bump 😮

File: packages/steelpenny/package.json [hunk 15]
```diff
@@ -43,16 +43,14 @@
...
     "pouchy": "2.0.0",
     "qs": "5.1.0",
     "redis": "1.0.0",
     "relations": "0.5.0",
     "url": "0.11.0",
     "uuid": "2.0.1",
-    "inject-then": "2.0.5",
-    "vision": "3.0.0",
-    "pg": "6.0.0"
```
Comment (rssk) [inline packages/steelpenny/package.json]: wat?

File: packages/steelpenny/.nycrc [hunk 16]
```diff
@@ -7,11 +7,12 @@
     "node_modules",
     "test"
   ],
-  "functions": 90,
+  "functions": 75,
```
Comment (rssk) [inline packages/steelpenny/.nycrc]: 😿

File: packages/steelpenny/package.json (hunk 14 above)
Comment (swashcap) [inline packages/steelpenny/package.json]: Yeah, it looks like `pg` is [included _twice_ as a dependency](https://github.com/MRN-Code/quarterback/blob/d4bbe6473d356d28a6e92af9b222c44be3791f99/packages/steelpenny/package.json#L46-L55)? npm just went with the more recent one.

File: packages/steelpenny/src/lib/utils/get-plugins.js [hunk 17]
```diff
@@ -5,13 +5,13 @@ const path = require('path');
 const pkg = require(path.join(__dirname, '../../../package.json'));
 const cliOpts = require('./cli-options.js');
 const chalk = require('chalk');
-const baseRoutePath = path.join(__dirname, '../app-routes/');
+const baseRoutePath = path.join(__dirname, '../routes/');
```
Comment (rssk) [inline packages/steelpenny/src/lib/utils/get-plugins.js]: 🙏

File: packages/steelpenny/src/lib/utils/get-plugins.js [hunk 18]
```diff
@@ -108,14 +108,14 @@ const plugins = [
...
+  getRouteConfig('auth/cookies.js', 'auth'),
+  getRouteConfig('auth/reset.js', 'auth'),
+  getRouteConfig('users.js'),
+  getRouteConfig('scans.js'),
+  getRouteConfig('scan-details.js'),
+  getRouteConfig('sites.js'),
+  getRouteConfig('dataexchange.js'),
   {
     register: require('./version-route.js'),
     registrationOptions: {
```
Comment (rssk) [inline packages/steelpenny/src/lib/utils/get-plugins.js]: So hot

File: packages/steelpenny/package.json (hunk 15 above)
Comment (swashcap) [inline packages/steelpenny/package.json]: Um…yeah. Not sure why this happened. ¯\_(npm)_/¯

File: packages/steelpenny/src/lib/utils/authentication.js (hunk 10 above)
Comment (rssk) [inline packages/steelpenny/src/lib/utils/authentication.js]: Cool, we should enforce this in reviews when the linter doesn't as well (ie: internals.*)

File: packages/steelpenny/test/integration/response-formatter.js (hunk 11 above)
Comment (rssk) [inline packages/steelpenny/test/integration/response-formatter.js]: https://www.npmjs.com/package/eslint-plugin-bdd?

File: packages/steelpenny/package.json (hunk 15 above)
Comment (swashcap) [inline packages/steelpenny/package.json]: Ah, looks like npm alpha-sorts keys. Dependencies are still included!

**Comment by swashcap** [comment]
Yay!

**Pull request merged**